daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -155,7 +164,9 @@ object Node {
       exerciseResult: Option[Value],
       key: Option[KeyWithMaintainers],
       override val byKey: Boolean, // invariant (!byKey || exerciseResult.isDefined)
-      // For the sake of consistency between types with a version field, keep this field the last.
+                                   // For the sake of consistency between types with a version field, keep this field the last.
+      byImplementation: Option[Implementation], // The implementation of the interface choice if any.
+      // This is set iff interfaceId is set and choiceId points to an abstract choice.
```

Comment:
The design says that templates can have abstract choices as well?

Reply:
Looks like I forgot to update the requirements, the actual design afterwards does not include that and imho it shouldn’t.

Reply:
Looks like I forgot to update the requirements, that definitely was not intended.